pkg/controllers/workspace_controller.go

```diff
@@ -827,16 +800,45 @@ func (c *WorkspaceReconciler) applyInference(ctx context.Context, wObj *kaitov1a
 				existingObj = &appsv1.Deployment{}
 
 			}
-
+			revisionStr := wObj.Annotations[kaitov1alpha1.WorkspaceRevisionAnnotation]
 			if err = resources.GetResource(ctx, wObj.Name, wObj.Namespace, c.Client, existingObj); err == nil {
 				klog.InfoS("An inference workload already exists for workspace", "workspace", klog.KObj(wObj))
+				switch deployment := existingObj.(type) {
+				case *appsv1.Deployment: // Only initcontainer, env and annotations change will be updated
+					var volumes []corev1.Volume
+					var volumeMounts []corev1.VolumeMount
+					shmVolume, shmVolumeMount := utils.ConfigSHMVolume(*wObj.Resource.Count)
+					if shmVolume.Name != "" {
+						volumes = append(volumes, shmVolume)
+					}
+					if shmVolumeMount.Name != "" {
+						volumeMounts = append(volumeMounts, shmVolumeMount)
+					}
+
+					if len(wObj.Inference.Adapters) > 0 {
+						adapterVolume, adapterVolumeMount := utils.ConfigAdapterVolume()
+						volumes = append(volumes, adapterVolume)
+						volumeMounts = append(volumeMounts, adapterVolumeMount)
+					}
+					initContainers, envs := resources.GenerateInitContainers(wObj, volumeMounts)
+					spec := &deployment.Spec
+					spec.Template.Spec.InitContainers = initContainers
+					spec.Template.Spec.Containers[0].Env = envs
+					spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
+					deployment.Annotations[kaitov1alpha1.WorkspaceRevisionAnnotation] = revisionStr
+					spec.Template.Spec.Volumes = volumes
+
+					if err := c.Update(ctx, deployment); err != nil {
```

Comment:
Tests needed for testing update mounts

Reply:
e.g. test initial volume mounts are correct, apply update, test updated volume mounts are correct